src/raw/props.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{spec::BinarySubtype, Binary, Bson, Document, JavaScriptCodeWithScope, Regex};
```

Comment:
I'm not sure I quite understand exactly what this is doing, could we add some comments inline to explain? Also, can we move this to the `test` module when created?

Reply:
proptests generate a large number of random inputs, run tests against those inputs, then minimize the input until it finds the smallest failing example it can.  These are the functions for generating arbitrary BSON objects, so proptest can verify that arbitrary BSON round-trips properly.  

Further reading:
* [A blog post introducing property testing in rust](https://blog.logrocket.com/property-based-testing-in-rust-with-proptest/)
* [The guide to the proptest library](https://altsysrq.github.io/proptest-book/intro.html)

Reply:
Very cool! This seems like a great way to test this.

Reply:
See also [`quickcheck`](https://hackage.haskell.org/package/QuickCheck) for info on the Haskell library that popularized this idea

Reply:
Does the prop definitions need to be in the actual library? Or can we move them into the `test` module?